neuron_morphology/pipeline/post_data_to_s3.py

```diff
@@ -0,0 +1,118 @@
+import os
+import boto3
+import json
+import configparser
+from zipfile import ZipFile, ZIP_DEFLATED
+from io import BytesIO
+from argschema import ArgSchemaParser
+from neuron_morphology.pipeline._schemas import InputParameters
+
+def zip_files(file_dict):
+    """
+    zip files into an archive in memory
+
+    Parameters
+    ---------------
+    file_dict: file name: file paths or file in bytes to be archived
+
+    Return
+    ---------------
+    archive: BytesIO obj
+
+    """
+
+    archive = BytesIO()
+
+    with ZipFile(archive, "a", ZIP_DEFLATED, False) as zip_file:
+        for name, file in file_dict.items():
+            if isinstance(file, str):
+                with open(file, 'rb') as fh:
+                    data = BytesIO(fh.read())
+                zip_file.writestr(name, data.getvalue())
+            elif isinstance(file, BytesIO):
+                zip_file.writestr(name, file.getvalue())
+            else:
+                raise TypeError("Invalid input file!")
+
+    return archive
+
+
+def post_object_to_s3(archive_data, archive_name, bucket, region, access_key_id=None, secret_access_key=None):
+    """
+    This zip files to an archive in memory and post it to S3 bucket
+
+    Parameters
+    ------------------
+    archive_data: the archive data
+    archive_name: the archive's name in s3
```

Comment:
these are not necessarily archives, could be a regular file

Reply:
revised.